code/modules/vehicles/engine_urist.dm

```diff
@@ -0,0 +1,200 @@
+ *  - Plugging an cell into an electric engine now has a chat message
+ *  - Fixed a code error where thermal overrides electrical "putter"
+ *  - Refactored thermal's fuel processing into its own proc
+ *  - Refactored thermal's fuel checks into several procs
+ *  - Added a start-up check proc; "can_turn_on()"
+ *  - Added "dead_start()" sound proc for failing start-ups
+ *  - Added status checking to 'power_use' and 'can_start_up'
+ *
+ */
+
+/obj/item/weapon/engine
+	name = "engine"
+	desc = "An engine used to power a small vehicle."
+	icon = 'icons/urist/items/vehicle_parts.dmi'
+	w_class = ITEM_SIZE_HUGE
+	var/stat = 0
+	var/trail_type
+	var/cost_per_move = 5
+
+/obj/item/weapon/engine/proc/get_trail()
+	if(trail_type)
+		return new trail_type
+	return null
+
+/obj/item/weapon/engine/proc/prefill()
+	return
+
+/obj/item/weapon/engine/proc/use_power()
+	return 0
+
+/obj/item/weapon/engine/proc/can_turn_on()
+	return 0
+
+/obj/item/weapon/engine/proc/dead_start(var/atom/movable/M, var/mob/user = null)
+	return
+
+/obj/item/weapon/engine/proc/rev_engine(var/atom/movable/M, var/mob/user = null)
+	return
+
+/obj/item/weapon/engine/proc/putter(var/atom/movable/M, var/mob/user = null)
+	return
+
+/obj/item/weapon/engine/electric
+	name = "electric engine"
+	desc = "A battery-powered engine used to power a small vehicle."
+	icon_state = "engine_electric"
+	trail_type = /datum/effect/effect/system/trail/ion
+	cost_per_move = 200	// W
+	var/obj/item/weapon/cell/cell
+
+/obj/item/weapon/engine/electric/attackby(var/obj/item/I, var/mob/user)
+	if(istype(I,/obj/item/weapon/cell))
+		if(cell)
+			to_chat(user, "<span class='warning'>There is already a cell in \the [src].</span>")
+		else
+			if (user && user.unEquip(I))
+				cell = I
+				I.forceMove(src)
+				to_chat(user, "You insert \the [cell] into \the [src].")
+		return 1
+	else if(isCrowbar(I))
+		if(cell)
+			to_chat(user, "You pry out \the [cell].")
+			cell.forceMove(get_turf(src))
+			cell = null
+			return 1
+	..()
+
+/obj/item/weapon/engine/electric/prefill()
+	cell = new /obj/item/weapon/cell/high(src.loc)
+
+/obj/item/weapon/engine/electric/use_power()
+	if(!cell || stat)
+		return 0
+	return cell.use(cost_per_move * CELLRATE)
+
+/obj/item/weapon/engine/electric/can_turn_on()
+	return !stat && cell && cell.check_charge(cost_per_move * CELLRATE)
+
+/obj/item/weapon/engine/electric/dead_start(var/atom/movable/M, var/mob/user)
+	if(user)
+		to_chat(user, "<i>You try to start the [M], but nothing happens.</i>")
+
+/obj/item/weapon/engine/electric/rev_engine(var/atom/movable/M)
+	M.audible_message("\The [M] beeps, spinning up.")
+
+/obj/item/weapon/engine/electric/putter(var/atom/movable/M)
+	M.audible_message("\The [M] makes one depressed beep before winding down.")
+
+/obj/item/weapon/engine/electric/emp_act(var/severity)
+	if(cell)
+		cell.emp_act(severity)
+	..()
+
+/obj/item/weapon/engine/electric/examine(mob/user)
+	..(user)
```

Comment:
Check if the user was able to examine before giving them any custom examine text.